### ISSUE TITLE
Linux IOM connection fails with hostname length 52 characters.

### ISSUE THREAD

**Issue by sas-rocks** [sassoftware/vscode-sas-extension#861]
**Describe the bug**
Trying to create a Linux IOM bridge connection I get the following error when the target hostname has 52 characters in length:
####
Error: There was an error executing the SAS Program. See [console log](command:workbench.action.toggleDevTools) for more details.
	at Hx.onShellStdErr (c:\Users\gertro\.vscode\extensions\sas.sas-lsp-1.7.1\client\dist\node\extension.js:264:155)
	at Socket.emit (node:events:514:28)
	at addChunk (node:internal/streams/readable:324:12)
	at readableAddChunk (node:internal/streams/readable:297:9)
	at Socket.push (node:internal/streams/readable:234:10)
	at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
####

The DeveloperTools report this in the console (I replaced the fully qualified hostname with a matching pattern):
####
[Extension Host] shellProcess stderr: <?xml version="1.0" ?><Exceptions><Exception><SASMessage severity="Error">Physical file does not exist, /tmp/SAS_workEC
7400100052_mys94serv.testxx-clusterx.demodom.intcloudx.sas.com/SAS_work0D7300100052_mys94serv.testxx-clusterx.demodom/e6ab14c7-
b1f3-4198-94d2-cc69ba104af3.htm.</SASMessage></Exception></Exceptions>
####
truncating the internal workfolder name in the second folder level by that part of the folder name ".intcloudx.sas.com".

With a hostname having only 20 characters the exact same configuration works fine as the work folder names are also shorter. Therefor it seems there's an internal length limit for that work folder name in the plug-in

**Steps to reproduce**
You need a SAS 9.4 environment (tested on M7 and M8) with a hostname length like "mys94serv.testxx-clusterx.demodom.intcloudx".
Define the connection "SAS: Add New Connection Profile" 
Use "SAS 9.4 (remote - IOM)
Enter the fully qualified long hostname (or the IP - makes no difference)
Add the userid and logon when running this sample code:
####
data test;
    set sashelp.class;
run;
####

**Expected behavior**
Working exactly the same way as the same code executed on a hostname with for example 20 characters.

**Screenshots**
![image](https://github.com/sassoftware/vscode-sas-extension/assets/58768304/1ea14a87-c7fa-40b9-bfcf-0f54556276a9)

**Environment (please complete the following information):**
Client OS: Windows 10
Extension version: 1.7.1
SAS version: 9.4 M7 & M8

**Additional context**

**Comment by clangsmith** [comment]
I tried to reproduce by mapping the IP for the server to a longer host name via the Windows hosts file (on the client side, so I didn't have to actually change the hostname of the Linux server) using the following steps, but it worked fine:

1. I started a Linux workspace server via the following command: 
/sas/tools/com/sdssas -t dev/mva-v940m8 -box laxnd -objectserver -objectserverparms "port=6339 lockserver nosecurity protocol=bridge" -metaserver uitsrv05.na.sas.com -metaport 8561
2. The name of the host that the session was launched on was something like "lax94m8d02.unx.sas.com".
3. I pinged that host to get the IP address.
4. On the client machine with VS Code, I then opened the Windows hosts file and added a line similar to:
11.11.1.11 mys94serv.testxx-clusterx.demodom.intcloudx
(where the IP is the IP address to the host where the Linux session is running found by pinging above)
5. I used "mys94serv.testxx-clusterx.demodom.intcloudx" as the hostname and 6339 as the port in my SAS Extension for VS Code profile.
6. I was still able to connect and submit code without problems, even with the mapped longer host name.

I'm now wondering if mapping the hostname on the Linux host/server might allow it to reproduce. Or may be easier to just find a Linux system we can change the hostname on.

**Comment by scottdover** [comment]
Hey @clangsmith. I took a look into this to see if there was anything else I could gleam, and I was unable to reproduce

**Comment by clangsmith** [comment]
I am able to reproduce with the (M8) environment @sas-rocks provided (thanks!) and I see the truncation you mentioned. The first line below is the path that comes back in the “Physical file does not exist” exception/error and the second line is the actual physical path:

/tmp/SAS_workA7CC000CD097_sasserver.tsgfdm-cloudera.sashq-d.openstack.sas.com/SAS_work0410000CD097_sasserver.tsgfdm-cloudera/a7067cbe-abdd-456f-8c5b-dfd14bd7ed98.htm

/tmp/SAS_workA7CC000CD097_sasserver.tsgfdm-cloudera.sashq-d.openstack.sas.com/SAS_work0410000CD097_sasserver.tsgfdm-cloudera**.sashq-d.openstack.sas.com**/a7067cbe-abdd-456f-8c5b-dfd14bd7ed98.htm

The part in bold is not in the exception text, so it does look like it is getting truncated at some point. I am able to open that same .htm in EG, so EG is successfully downloading the file.

**Comment by scottdover** [comment]
Hey @sas-rocks/@clangsmith . This issue will be resolved as part of https://github.com/sassoftware/vscode-sas-extension/pull/842 and will be released with 1.8.0. 

Currently, we rely on log parsing to get information about our SAS 9 session. We get this information line by line. So, the work directory for this server was showing up like...

```
line 1: WORKDIR=/tmp/<beginningOfThePath>
lline 2: <endOfThePath>
```

This is why we were seeing the truncation. I've since updated our parser to capture the full path and things seem to be working as expected. 

Thanks for bringing this to our attention, @sas-rocks! I'm for the inconvenience this has caused

**Comment by clangsmith** [comment]
Just to follow-up, this issue was *not* fully resolved as part of #842 as previously thought. @sas-rocks reported and I confirmed that the problem still occurred with v1.9.0. We are now confident this is fixed with #978, since @scnwwu explicitly confirmed his changes for #978 fixed this scenario (using an environment @sas-rocks provided, thanks!).
